datadog/resource_datadog_security_monitoring_rule.go

```diff
@@ -412,7 +412,7 @@ func datadogSecurityMonitoringRuleSchema() map[string]*schema.Schema {
 		"filter": {
 			Type:        schema.TypeList,
 			Optional:    true,
-			Description: "Additional queries to filter matched events before they are processed.",
+			Description: "Additional queries to filter matched events before they are processed. **Note**: This field is deprecated for log detection, signal correlation, and workload security rules.",
```

Comment:
```suggestion
			Description: "Additional queries to filter matched events before they are processed.,
			Deprecated: "This field is deprecated for log detection, signal correlation, and workload security rules.",
```

Reply:
Will this display a warning every time the field is used? Asking because it's deprecated only when used in conjunction with specific values for another field

Reply:
Ah I see thanks for the clarification. I thought this was deprecated in favor of the 3 listed.